datashuttle/configs/canonical_configs.py

```diff
@@ -18,28 +18,27 @@
 from pathlib import Path
 from typing import Literal, Union, get_args, get_origin
 
-from datashuttle.utils import utils
+from datashuttle.utils import folders, utils
 from datashuttle.utils.custom_exceptions import ConfigError
 
 
-# TODO: this can be merged with get_canonical_config_required_types()
-def get_canonical_config_dict() -> dict:
+def get_canonical_configs() -> dict:
     """
-    Canonical list of required keys
-    in the datashuttle configs
+    The only permitted types for DataShuttle
+    config values.
     """
-    config_dict = {
-        "local_path": None,
-        "central_path": None,
-        "connection_method": None,
-        "central_host_id": None,
-        "central_host_username": None,
-        "overwrite_old_files": None,
-        "transfer_verbosity": None,
-        "show_transfer_progress": None,
+    canonical_configs = {
+        "local_path": Union[str, Path],
```

Comment:
since you use this type a lot, you may benefit from defining a custom type alias:

```python
StrOrPath = Union[str, Path]
```
for Python >= 3.10:
```python
StrOrPath: TypeAlias = Union[str, Path]
```

In this particular example, probably it doesn't save you much space, but keep it in mind if the need arises.

Reply:
Thanks! I will keep this in mind.